libsolidity/interface/Natspec.cpp

```diff
@@ -100,15 +100,52 @@ Json::Value Natspec::devDocumentation(ContractDefinition const& _contractDef)
 		{
 			Json::Value method(devDocumentation(fun->annotation().docTags));
 			if (!method.empty())
+			{
 				// add the function, only if we have any documentation to add
+				Json::Value ret(Json::objectValue);
+
+				auto returnParams = fun->returnParameters();
+				auto returnDoc = fun->annotation().docTags.equal_range("return");
+
+				if (!returnParams.empty())
+				{
+					unsigned int n = 0;
```

Comment:
Since your incrementing `n` in the loop body in every case, you could remove this variable and use `i` for placeholder creation. Or is `returnDoc.first` returning a value > 0?

Reply:
`i` is a map iterator; maybe I'm missing something on how to use that to access the vector for the return parameters. 😅 

Would changing the type for `n` to`size_t` help?

Reply:
Oh, true. You're right. We need to iterate the map. I think the implementation is fine then and just needs some minor cleanup. One last proposal: maybe move the extraction into a function?
```
Json::Value Natspec::extractReturnParameterDocs(std::multimap<std::string, DocTag> const& _tags, FunctionDefinition const& _functionDef)
{
    if (!_functionDef.returnParameters().empty())
    {
        size_t index = 0;
        for (auto i = returnDoc.first; i != returnDoc.second; i++)
        {
            string paramName = _functionDef.returnParameters().at(n)->name();
            string content = i->second.content;
            ...
        }
    }
}
```
And then call it here:
```
if (!method.empty())
{
    auto returnDoc = fun->annotation().docTags.equal_range("return");
    
    Json::Value returnParameterDocs = extractReturnParameterDocs(returnDoc, *fun);
    
    if (!returnParameterDocs.empty())
	    method["return"] = returnParameterDocs;
    ....
}
```